### PR TITLE
Fix/postal codes only accept brazil format

### PR DESCRIPTION
**Pull request by thayannevls**
#### What is the purpose of this pull request?
Change shipping-simulator to use [`address-form`](https://github.com/vtex/address-form) and apply the correct postal code rules for the current country.

*Also:* Now shipping-simulator is using hooks.

#### What problem is this solving?
Shipping-simulator was only validating postal code with the Brazil format(8 digits) and not checking the current country.

#### How should this be manually tested?
[Workspace](https://thayy--storecomponents.myvtex.com/traveler-backpack/p)

**Examples**
Valid postal code: 10002
Invalid: 32132111

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ShippingSimulator/Loader.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import ContentLoader from 'react-content-loader'
+
+import shippingSimulator from './shippingSimulator.css'
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/Loader.js]: nit:
```suggestion
import styles from './shippingSimulator.css'
```

File: react/components/ShippingSimulator/components/ShippingTable.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { FormattedMessage } from 'react-intl'
 
 import ShippingTableRow from './ShippingTableRow'
 
-import styles from '../styles.css'
+import shippingSimulator from '../shippingSimulator.css'
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/components/ShippingTable.js]:
```suggestion
import styles from '../shippingSimulator.css'
```

File: react/components/ShippingSimulator/components/ShippingTableRow.js [hunk 3]
```diff
@@ -4,18 +4,18 @@ import { intlShape, injectIntl } from 'react-intl'
 import TranslateEstimate from 'vtex.shipping-estimate-translator/TranslateEstimate'
 import classNames from 'classnames'
 
-import styles from '../styles.css'
+import shippingSimulator from '../shippingSimulator.css'
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/components/ShippingTableRow.js]:
```suggestion
import styles from '../shippingSimulator.css'
```

File: react/components/ShippingSimulator/index.js [hunk 4]
```diff
@@ -1,97 +1,54 @@
...
-      zipcodeValue = zipcodeValue.substring(0, 8)
-    }
-
-    if (zipcodeValue.length > 5) {
-      zipcodeValue = zipcodeValue.replace(/(\d{5})(\d+)/, '$1-$2')
-    }
-
-    this.setState({ zipcodeValue })
+import ShippingSimulatorLoader from './Loader'
+import shippingSimulator from './shippingSimulator.css'
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/index.js]:
```suggestion
import styles from './shippingSimulator.css'
```

File: react/components/ShippingSimulator/index.js [hunk 5]
```diff
@@ -1,97 +1,54 @@
...
+
+const ShippingSimulator = ({
+  intl,
+  client,
+  skuId,
+  seller,
+  country,
+  styles,
+}) => {
+  const [address, setAddress] = useState(addValidation(getNewAddress(country)))
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/index.js]: use the lazy initialization, so you don't call `getNewAddress` on every render
```suggestion
  const [address, setAddress] = useState(() => addValidation(getNewAddress(country)))
```

File: react/components/ShippingSimulator/index.js [hunk 6]
```diff
@@ -1,97 +1,54 @@
...
+  const [shipping, setShipping] = useState({})
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/index.js]: why not initialize with null?

File: react/components/ShippingSimulator/index.js [hunk 7]
```diff
@@ -1,97 +1,54 @@
...
+import shippingSimulator from './shippingSimulator.css'
+import { getNewAddress } from './utils'
+
+const ShippingSimulator = ({
+  intl,
+  client,
+  skuId,
+  seller,
+  country,
+  styles,
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/index.js]: maybe rename this to `loaderStyles`? do you know if anyone is using this?

File: react/components/ShippingSimulator/utils/index.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+let addressId = 1
+
+const getRandomAddressId = () => {
+  return (addressId++ * new Date().getTime() * -1).toString().replace('-', '')
```
Comment (vitoriaHeliane) [inline react/components/ShippingSimulator/utils/index.js]: You can write this function this way:

```
const getRandomAddressId = () => (addressId++ * new Date().getTime() * -1).toString().replace('-', '')
```

File: react/components/ShippingSimulator/utils/index.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export const getNewAddress = country => {
+  const addressId = getRandomAddressId()
+  return {
+    addressId: addressId,
```
Comment (vitoriaHeliane) [inline react/components/ShippingSimulator/utils/index.js]:
```suggestion
    addressId,
```

File: react/components/ShippingSimulator/index.js [hunk 10]
```diff
@@ -102,66 +59,60 @@ class ShippingSimulator extends Component {
...
-        <ShippingTable shipping={shipping} />
-      </Fragment>
-    )
-  }
+            <PostalCodeGetter />
+          </AddressContainer>
+        </AddressRules>
+        <Button
+          onClick={handleClick}
+          className={`${shippingSimulator.shippingCTA}`}
```
Comment (vitoriaHeliane) [inline react/components/ShippingSimulator/index.js]:
```suggestion
          className={shippingSimulator.shippingCTA}
```

File: react/components/ShippingSimulator/index.js [hunk 11]
```diff
@@ -1,97 +1,54 @@
...
-    this.props.client
+    setLoading(true)
+    const { postalCode } = removeValidation(address)
+    client
       .query({
         query: getShippingEstimates,
         variables: {
           country,
-          postalCode: this.state.zipcodeValue,
+          postalCode: postalCode,
```
Comment (vitoriaHeliane) [inline react/components/ShippingSimulator/index.js]:
```suggestion
          postalCode,
```

File: react/components/ShippingSimulator/components/ShippingTableRow.js [hunk 12]
```diff
@@ -36,10 +36,10 @@ const ShippingTableRow = ({ name, shippingEstimate, price, intl }) => {
 
   return (
     <tr key={name}>
-      <td className={`${styles.shippingTableCell} pv1 ph3 t-small`}>
-        <label className={`${styles.shippingTableLabel}`}>
+      <td className={`${shippingSimulator.shippingTableCell} pv1 ph3 t-small`}>
+        <label className={`${shippingSimulator.shippingTableLabel}`}>
```
Comment (vitoriaHeliane) [inline react/components/ShippingSimulator/components/ShippingTableRow.js]:
```suggestion
        <label className={shippingSimulator.shippingTableLabel}>
```

File: react/components/ShippingSimulator/Loader.js (hunk 1 above)
Comment (thayannevls) [reply react/components/ShippingSimulator/Loader.js]: If I name *styles* it gets conflicted with the props @lucasecdb

File: react/components/ShippingSimulator/components/ShippingTableRow.js [hunk 13]
```diff
@@ -4,18 +4,18 @@ import { intlShape, injectIntl } from 'react-intl'
...
     {
       tc: shippingEstimate === undefined,
     }
   )
 
   const valueClassName = classNames(
-    `${styles.shippingTableCell} pv1 ph3 t-small c-muted-2`,
+    `${shippingSimulator.shippingTableCell} pv1 ph3 t-small c-muted-2`,
     {
       tc: price === undefined,
```
Comment (ArthurSampaio) [inline react/components/ShippingSimulator/components/ShippingTableRow.js]: I know that isnt your task, but its a small alteration.
If goal is know if a variable is `undefined` the correct way is more like `typeof foo === "undefined"`. This little mistake change the semantic of the command and can introduce unexpected behavoir. Can you check if this make sense in our case?

File: react/__mocks__/vtex.address-form/index.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { addValidation, removeValidation } from './helpers'
+import { StyleguideInput } from './inputs'
+
+const AddressRules = props => <div className="address-rules-mock" />
```
Comment (ArthurSampaio) [inline react/__mocks__/vtex.address-form/index.js]: Just a doubt. This element is used to hook something, right?

File: react/components/ShippingSimulator/Loader.js (hunk 1 above)
Comment (lucasecdb) [reply react/components/ShippingSimulator/Loader.js]: the prop is on the `ShippingSimulator`, no?

File: react/__mocks__/vtex.address-form/index.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+
+const AddressContainer = props => <div className="address-container-mock" />
```
Comment (ArthurSampaio) [inline react/__mocks__/vtex.address-form/index.js]: Use css modules 🔨 .

File: react/components/ShippingSimulator/components/ShippingTableRow.js (hunk 13 above)
Comment (thayannevls) [reply react/components/ShippingSimulator/components/ShippingTableRow.js]: It makes senses, just changed it!

File: react/__mocks__/vtex.address-form/index.js (hunk 15 above)
Comment (thayannevls) [reply react/__mocks__/vtex.address-form/index.js]: These `classNames`s are only used to make the snapshots better readable and not to customize the divs.

I checked in other tests and no one uses css modules, so I think that this confirms. Can you check again @ArthurSampaio ?

File: react/components/ShippingSimulator/Loader.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import ContentLoader from 'react-content-loader'
+
+import styles from './shippingSimulator.css'
+
+const ShippingSimulatorLoader = (loaderProps = {}) => (
+  <div className={`${styles.shippingContainer}`}>
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/Loader.js]: nit:
```suggestion
  <div className={styles.shippingContainer}>
```

File: react/components/ShippingSimulator/Loader.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    <ContentLoader
+      className={`${styles.shippingContainerLoader}`}
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/Loader.js]: nit:
```suggestion
      className={styles.shippingContainerLoader}
```

File: react/components/ShippingSimulator/Loader.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+      }}
+      width={500}
+      height={40}
+      preserveAspectRatio="xMinYMin meet"
+      {...loaderProps}
+    >
+      <rect
+        height="100%"
+        width="7em"
+        {...loaderProps[`${styles.shippingZipcodeLabelLoader}`]}
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/Loader.js]: nit:
```suggestion
        {...loaderProps[styles.shippingZipcodeLabelLoader]}
```

File: react/components/ShippingSimulator/Loader.js [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+      />
+      <rect
+        height="100%"
+        width="15em"
+        x="8em"
+        {...loaderProps[`${styles.shippingInputLoader}`]}
```
Comment (lucasecdb) [inline react/components/ShippingSimulator/Loader.js]: nit:
```suggestion
        {...loaderProps[styles.shippingInputLoader]}
```

**Pull request merged**